pinot-core/src/main/java/org/apache/pinot/core/operator/AcquireReleaseColumnsSegmentOperator.java

```diff
@@ -58,8 +58,8 @@ public AcquireReleaseColumnsSegmentOperator(PlanNode planNode, IndexSegment inde
    * Runs the planNode to get the childOperator, and then proceeds with execution.
    */
   @Override
-  protected IntermediateResultsBlock getNextBlock() {
-    _childOperator = (Operator<IntermediateResultsBlock>) _planNode.run();
+  protected BaseResultsBlock getNextBlock() {
+    _childOperator = (Operator<BaseResultsBlock>) _planNode.run();
     return _childOperator.nextBlock();
```

Comment:
unrelated to this PR but why are we running the plannode everytime to get child operator? i know it will access segment before running the plan, but does it change on a per segment bases?

Reply:
This is a workaround to acquire the segment before running the plan. Currently the plan is physical execution plan, which is per-segment based.